packages/react-router-dom/index.tsx

```diff
@@ -1246,7 +1267,7 @@ function useScrollRestoration({
 
       // been here before, scroll to it
       if (typeof restoreScrollPosition === "number") {
-        window.scrollTo(0, restoreScrollPosition);
+        scrollY(elementRef, restoreScrollPosition);
```

Comment:
This might be a _me_ problem, but when testing this with patch-package, I found I could only get this scroll restore to work by cheating:

```ts
setTimeout(() => scrollY(elementRef, restoreScrollPosition));
```

This is obviously a terrible thing to do, but it at least suggests what the problem is, namely that my scrolling element doesn't have its content yet when this tries to scroll it. I also tried changing the `useLayoutEffect` to a `useEffect` in case that might happen slightly later, but it didn't work. Curious if you have an experience with or insight into this, or if I just need to figure out how to get my stuff to render faster. In theory, because the content is fetched in a loader, everything should be showing up at the same time, but I guess it isn't.

Reply:
Ok, definitely a me problem. On a page in my app that doesn't render anything interesting, it works without the timeout. The problematic pages are using React Table, and they may be doing so in a suboptimal way. In general it's worth keeping an eye on though — if your scrollable content is not fully present in the DOM at navigation time, scroll restoration will not work.